reddit_api_async/connection.ml

```diff
@@ -13,6 +13,15 @@ module Credentials = struct
     [@@deriving sexp]
   end
 
+  module Refresh_token = struct
+    type t =
+      { client_id : string
+      ; client_secret : string option
+      ; refresh_token : string
+      }
+    [@@deriving sexp]
+  end
```

Comment:
@nmtake Could I trouble you to comment on this interface? I think you understand OAuth better than I do.

Reply:
No trouble. How does a user get the refresh token? Doesn't ocaml-reddit-api provide the way?

Reply:
No, thus far my goal is just to provide a way to use a refresh token that's assumed to come from somewhere else. (In the actual use case that's motivating this change, I have a bot that uses a refresh token that I just obtained from [PRAW's script](https://praw.readthedocs.io/en/latest/tutorials/refresh_token.html#obtaining-refresh-tokens). But of course, providing a way to get the refresh tokens via OCaml would be a good next step.

Reply:
I see. Should we add `[@sexp.option]` to the `Refresh_token.client_secret`, or remove the annotation from `Userless_public.device_id`?

Reply:
I vote for getting rid of it in `Userless_public`. I kinda dislike `[@sexp.option]` because it means that you can't infer the serialization from the interface.

Reply:
I'm happy with the change. It would suffice to not use `[@sexp.option]`.

Reply:
Thanks!